src/subplots.ts

```diff
@@ -1358,10 +1477,523 @@ export class PrimitiveGroupContainer extends PlotData {
 }
 
 
-export class Histogram extends PlotData {
+export class BasePlot extends PlotData {
+  public axes: newAxis[] = [];
+  public origin: Vertex;
+  public size: Vertex;
+  public translation: Vertex = new Vertex(0, 0);
+
+  public hoveredIndex: number[] = [];
+  public clickedIndex: number[] = [];
+  public selectedIndex: boolean[];
+
+  public viewPoint: Vertex = new Vertex(0, 0);
+  public fixedObjects: any[] = [];
+  public movingObjects: any[] = [];
+
+  protected initScale: Vertex = new Vertex(1, -1);
+  private _axisStyle = new Map<string, any>([['strokeStyle', 'hsl(0, 0%, 31%)']]);
+
+  readonly features: Map<string, any[]>;
+  readonly MAX_PRINTED_NUMBERS = 16;
+  readonly TRL_THRESHOLD = 20;
+  constructor(
+    public data: any,
+    public width: number,
+    public height: number,
+    public buttons_ON: boolean,
+    public X: number,
+    public Y: number,
+    public canvas_id: string,
+    public is_in_multiplot: boolean = false
+    ) {
+      super(data, width, height, buttons_ON, X, Y, canvas_id, is_in_multiplot);
+      this.origin = new Vertex(0, 0);
+      this.size = new Vertex(width - X, height - Y);
+      this.features = this.unpackData(data);
+      this.selectedIndex = Array.from([...this.features][0][1], x => x = false);
+      this.scaleX = this.scaleY = 1;
+      this.TRL_THRESHOLD /= Math.min(Math.abs(this.initScale.x), Math.abs(this.initScale.y));
+      this.refresh_MinMax();
+    }
+
+  refresh_MinMax(): void {
+    this.minX = this.origin.x;
+    this.maxX = this.origin.x + this.size.x;
+    this.minY = this.origin.y;
+    this.maxY = this.origin.y + this.size.y;
+  }
+
+  set axisStyle(newAxisStyle: Map<string, any>) {newAxisStyle.forEach((value, key) => this._axisStyle.set(key, value))}
+
+  get axisStyle() {return this._axisStyle};
+
+  get canvasMatrix() {return new DOMMatrix([this.initScale.x, 0, 0, this.initScale.y, this.origin.x, this.origin.y])}
+
+  get movingMatrix() {return new DOMMatrix([this.initScale.x, 0, 0, this.initScale.y, this.origin.x, this.origin.y])}
```

Comment:
moving and canvas matrices seem exactly the same. Is it supposed to ?

Reply:
Here yes, because there is no frame,
movingMatrix is overWritten in frame and in any further moving plot